delfin/drivers/dell_emc/vmax/client.py

```diff
@@ -287,6 +287,109 @@ def list_volumes(self, storage_id):
             LOG.error(msg)
             raise exception.StorageBackendException(msg)
 
+    def list_controllers(self, storage_id):
+        try:
+            # Get list of Directors
+            directors = self.rest.get_director_list(self.array_id,
+                                                    self.uni_version)
+            controller_list = []
+            for director in directors:
+                director_info = self.rest.get_director(
+                    self.array_id, self.uni_version, director)
+
+                status = constants.ControllerStatus.NORMAL
+                if director_info.get('availability', '').upper() != 'ONLINE':
+                    status = constants.ControllerStatus.OFFLINE
+
+                controller = {
+                    'name': 'director_'
+                            + str(director_info.get('director_number')),
+                    'storage_id': storage_id,
+                    'native_controller_id': director_info['directorId'],
+                    'status': status,
+                    'location':
+                        'slot_' +
+                        str(director_info.get('director_slot_number')),
+                    'soft_version': None,
+                    'cpu_info': 'number_of_cores_'
```

Comment:
expecting a model here.. having 'number_of_cores_' here does not looks good for me..

Reply:
Changed to 'Cores-*'